### PR TITLE
warn on cyclical imports

### PR DESCRIPTION
**Pull request by sharmilajesupaul**
Adds [`import/no-cycle`](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-cycle.md) rule to `airbnb-config-base`

### CODE REVIEW

**Comment by lencioni** [comment]
Do we want to set maxDepth?

**Comment by sharmilajesupaul** [comment]
@lencioni seems like by default, the `maxDepth` is set to `Infinity`. Is there a certain depth we want to allow? or should I explicitly set the `maxDepth` to the default in our config, so the behavior doesn't change if the default is changed?

File: packages/eslint-config-airbnb-base/rules/imports.js [hunk 1]
```diff
@@ -226,6 +226,10 @@ module.exports = {
     // https://github.com/benmosher/eslint-plugin-import/blob/44a038c06487964394b1e15b64f3bd34e5d40cde/docs/rules/no-self-import.md
     'import/no-self-import': 'error',
 
+    // Forbid cyclical dependencies between modules
+    // https://github.com/benmosher/eslint-plugin-import/blob/d81f48a2506182738409805f5272eff4d77c9348/docs/rules/no-cycle.md
+    'import/no-cycle': 'error',
```
Comment (ljharb) [inline packages/eslint-config-airbnb-base/rules/imports.js]: please do always hardcode the defaults (Infinity is fine)

**Comment by sharmilajesupaul** [comment]
ok updated @ljharb

**Pull request merged**